Sources/GRPC/Interceptor/ClientParts.swift

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2020, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import NIOHPACK
+
+public enum ClientRequestPart<Request> {
+  /// User provided metadata sent at the start of the request stream.
+  case metadata(HPACKHeaders)
+
+  /// A message to send to the server.
+  case message(Request, compress: Bool, flush: Bool)
```

Comment:
`compress` and `flush` seem like two of a possibly-extendable set of things we might want to attach to a message. I can in principle believe that `flush` should be promoted to top-level, but should this have a `Metadata` struct instead of `compress`?

Reply:
Yeah, that's a good call I think.